rudder-web/src/main/scala/com/normation/rudder/web/services/Section2FieldService.scala

```diff
@@ -223,41 +220,40 @@ class Section2FieldService(val fieldFactory: DirectiveFieldFactory, val translat
     SectionFieldImp(sectionSpec.name, children, priorityToVisibility(sectionSpec.displayPriority), varMappings)
   }
 
-
-  // transforms
-  // Map(A -> Seq("A1", "A2"), B -> Seq("B1", "b2"))
-  // to
-  // Seq( Map((A -> "A1"), (B -> "B1")),
-  //      Map((A -> "A2"), (B -> "B2")) )
-  //If there is no value, a None is returned
+  // Transforms map of all values by Variable Name
+  // Into the Sequence of all values by section
+  // A section will be defined as a Map of value by variable name
+  //   * ie: Map(A -> Seq("A1", "A2"), B -> Seq("B1", "B2"))  => Seq( Map((A -> "A1"), (B -> "B1"))
+  //                                                                , Map((A -> "A2"), (B -> "B2")) )
+  // If there is no value, a Section with default value is returned is returned
   private def createMapForEachSubSection(section: SectionSpec, valuesByName:Map[String,Seq[String]]): Seq[Map[String, Option[String]]] = {
-    // values represent all the values we have for the same name of variable
-    case class NameValuesVar(name: String, values: Seq[String]) extends HashcodeCaching
 
-    // seq of variable values with same name correctly ordered
-    val seqOfNameValues : Seq[NameValuesVar] = {
-      for {
-        varSpec <- section.getAllVariables
-      } yield {
-        NameValuesVar(varSpec.name, valuesByName.getOrElse(varSpec.name, Seq[String]()))
-      }
-    }
+    // Variable with max value determine the actual number of section
+    // If a varaible has 6 values, We need to have 6 sections at the end
```

Comment:
Are you sure that holds? What about variables in sub-sections ? (perhaps it holds only because we don't have multi-valued sub section of multi-valued section, is it ?)